### PR TITLE
Remove dead code (related to stochastic function) + clean up input buffer

### PR DESCRIPTION
**Pull request by apaszke**
As in the title. Next commits will remove `is_executable`, since it shouldn't be necessary anymore

### CODE REVIEW

**Comment by apaszke** [comment]
I removed the changes related to input buffer, because VariableType can't really handle sparse additions

File: torch/autograd/variable.py [hunk 1]
```diff
@@ -432,7 +432,11 @@ def __dir__(self):
     class _torch(object):
         @staticmethod
         def normal(means, std=1):
-            return Normal.apply(means, std)
+            if isinstance(means, Variable):
```
Comment (fmassa) [inline torch/autograd/variable.py]: I don't know how was the behavior before, but this means that one can't learn the `mean`/`std` parameters that are used to sample from a normal distribution?

Comment (apaszke) [reply torch/autograd/variable.py]: It wasn’t reparsmetrized before this PR, so you couldn’t do it anyway. The only mechanism that would let you backprop to mean and std is reinforce, which was replaced by log_prob method

**Review by colesbury — APPROVED** [review]
lgtm

**Pull request merged**